common/src/main/java/com/wynntils/screens/maps/MainMapScreen.java

```diff
@@ -336,17 +346,22 @@ public boolean doMouseClicked(double mouseX, double mouseY, int button) {
         }
 
         if (button == GLFW.GLFW_MOUSE_BUTTON_RIGHT) {
-            if (McUtils.player().isShiftKeyDown()
-                    && Models.Compass.getCompassLocation().isPresent()) {
-                Location location = Models.Compass.getCompassLocation().get();
+            List<MarkerInfo> markers =
+                    Models.Marker.USER_WAYPOINTS_PROVIDER.getCompassInfos().toList();
+            if (McUtils.player().isShiftKeyDown() && !markers.isEmpty()) {
+                int index = markers.indexOf(focusedMarker);
+                MarkerInfo markerInfo = markers.get((index + 1) % markers.size());
+                focusedMarker = markerInfo;
+                Location location = markerInfo.location();
```

Comment:
Is this the right behavior, to both center and switch to the next marker in the same operation? Maybe it is, I dunno. Does it feel natural when you do this?

Reply:
This feels okay imo, we focus on the next one with a click.